packages/components/src/internal/url/URLResolver.ts

```diff
@@ -243,6 +245,34 @@ const ASSAY_MAPPERS = [
             }
         }
     }),
+
+    // Issue 52151: resolve assay runs from search results
+    new ActionMapper('experiment', 'showRunGraph', row => {
+        const url = row.get('url');
+        if (url) {
+            const hit: SearchHit = row.toJS();
+            if (hit.category === SearchCategory.AssayRun) {
+                const runId = parseInt(hit.data?.id, 10);
+                if (!isNaN(runId)) {
+                    return AppURL.create('rd', 'assayrun', runId);
+                }
+            }
+        }
+    }),
+
+    // Issue 52151: resolve assay batches from search results
+    new ActionMapper('experiment', 'details', row => {
```

Comment:
Is this really only going to be an assay batch? `experiment-details` seems pretty generic. I don't see it used anywhere else in this context, but since the `DetailsAction` in the `ExperimentController` is for showing details of any run it just seems like it might be a false positives on this match (for workflow jobs, for example?). I don't have a better suggestion, though.

Reply:
No, it's not. Neither of these will be specific which is why I have the caveat of the search category. Thus, we'll only resolve a URL when it is `experiment-details` AND has a search category of `AssayBatch`.